File: web/vuejs/src/components/shared/font.ts
/**
 * Copyright (c) 2025 worldiety GmbH
 *
 * This file is part of the NAGO Low-Code Platform.
 * Licensed under the terms specified in the LICENSE file.
 *
 * SPDX-License-Identifier: Custom-License
 */
// see also https://developer.mozilla.org/en-US/docs/Web/CSS/font
import { Font, FontStyleValues } from '@/shared/proto/nprotoc_gen';

export function fontCSS(font?: Font): string[] {
	const styles: string[] = [];
	if (!font) {
		return styles;
	}

	// style and weight must precede size
	switch (font.style) {
		case FontStyleValues.Normal:
			styles.push('font-style: normal');
			break;
		case FontStyleValues.Italic:
			styles.push('font-style: italic');
			break;
	}

	if (font.weight) {
		styles.push(`font-weight: ${font.weight}`);
	}

	if (font.name) {
		styles.push(`font-family: ${font.name}`);
	}

	if (font.size) {
		styles.push(`font-size: ${font.size}`);
	}

	if (font.lineHeight) {
		styles.push(`line-height: ${font.lineHeight}`);
	}

	return styles;
}

File: presentation/core/font.go
// Copyright (c) 2025 worldiety GmbH
//
// This file is part of the NAGO Low-Code Platform.
// Licensed under the terms specified in the LICENSE file.
//
// SPDX-License-Identifier: Custom-License

package core

type FontFace struct {
	// Family is an arbitrary name for a font family to match against different variants.
	Family string `json:"family,omitempty"`

	// Style is like normal, italic, oblique
	Style string `json:"style,omitempty"`

	// Weight property of the FontFace interface retrieves or sets the weight of the font. Like 400 or 700 or bold
	// or normal.
	Weight string `json:"weight,omitempty"`

	// Source where to download the font.
	Source URI `json:"URL,omitempty"`
}

type Fonts struct {
	DefaultFont string     `json:"default,omitempty"`
	Faces       []FontFace `json:"faces,omitempty"`
}

func (ff Fonts) Contains(family string) bool {
	for _, face := range ff.Faces {
		if face.Family == family {
			return true
		}
	}
	
	return false
}
